patroni/dcs/__init__.py

```diff
@@ -1033,8 +1078,8 @@ def sync_state(leader: Union[str, None], sync_standby: Union[Collection[str], No
         """
         return {'leader': leader, 'sync_standby': ','.join(sorted(sync_standby)) if sync_standby else None}
 
-    def write_sync_state(self, leader: Union[str, None], sync_standby: Union[Collection[str], None],
-                         index: Optional[Union[int, str]] = None) -> bool:
+    def write_sync_state(self, leader: Optional[str], sync_standby: Optional[Collection[str]],
+                         index: Optional[_Version] = None) -> bool:
```

Comment:
```suggestion
                         index: Optional[Any] = None) -> bool:
```

All other occurrences of `index` parameter in methods of `AbstractDcs` class use `Optional[Any]`.
That leads me to believe this one should also follow the same convention.

Besides that, while taking a look at the `index` usage, it seems to expect an `int` which is incremented upon modifications. What is the usage of `str` for `index`?